### PR TITLE
[Index] Provide capability for the client to explicitly specify the list of output unit files for the datastore

### PR DESCRIPTION
**Pull request by akyrtzi**
The benefits of this approach is that it allows the datastore to ignore leftover unit files so it can both improve performance and correctness by not taking into account stale data.

rdar://61899059

### CODE REVIEW

**Comment by akyrtzi** [comment]
@swift-ci Please test

**Comment by akyrtzi** [comment]
@swift-ci Please test

**Comment by akyrtzi** [comment]
@swift-ci Please test Windows platform

File: lib/Index/IndexDatastore.cpp [hunk 1]
```diff
@@ -108,12 +116,16 @@ class StoreUnitRepo : public std::enable_shared_from_this<StoreUnitRepo> {
   mutable llvm::sys::Mutex StateMtx;
   std::unordered_map<IDCode, std::shared_ptr<UnitMonitor>> UnitMonitorsByCode;
 
+  llvm::StringSet<> ExplicitOutputUnitsSet;
```
Comment (benlangmuir) [inline lib/Index/IndexDatastore.cpp:119]: Does this need to be the names, or could it be the hashes?  This is going to be a really big set if it stores string data.

Comment (akyrtzi) [reply lib/Index/IndexDatastore.cpp:119]: Good point, will do that 👍

**Comment by akyrtzi** [comment]
@swift-ci Please test

**Comment by akyrtzi** [comment]
@swift-ci test linux platform

**Comment by akyrtzi** [comment]
@swift-ci test linux platform

**Pull request merged**